sirinternal.c

```diff
@@ -1122,6 +1124,10 @@ bool _sir_getthreadname(char name[SIR_MAXPID]) {
         _sir_handleerr(ret);
         return false;
     }
+# if defined(__HAIKU__)
+    if ((strcmp(name, "pthread_func")) || (name[0] == '\0'))
```

Comment:
can you please use strncmp(name, "pthread_func", SIR_MAXPID)? and you can use _sir_validstrnofail for the 2nd part; that's what it's for–checks null and first byte.